api/service/v1alpha1/service.proto

```diff
@@ -495,7 +495,7 @@ message UpdateCredentialsRequest {
   string name = 2;
   string type = 3;
   string repo_url = 4 [json_name = "repoURL"];
-  string repo_url_pattern = 5 [json_name = "repoURLPattern"];
+  bool repo_url_is_regex = 5 [json_name = "repoURLIsRegex"];
```

Comment:
If regex must be prefixed with `regex:`, is this boolean necessary?

Reply:
Granted, it might be a tiny bit leaky. Because now when building an edit/update modal, I suppose you have to go in the other direction and  _look_ for the "regex:" prefix to know whether the checkbox should be pre-checked or not...

Maybe I ought to keep this as a boolean all the way through to the secret instead of using a prefix ever/at all.

Reply:
What's the advantage of having to remember a field to check for a boolean over having to check a field value for a prefix?

Reply:
@hiddeco I think the advantage of a field over the prefix is IDE auto-complete and type checking, and simpler logic (no need to parse a string)... @krancour my opinion is to go with the boolean field and no prefix, but I don't feel strongly either way (except that we should choose one or the other and not both)

Reply:
I am happy to provide a boolean in the API for the reasons mentioned, but I think doing it all the way through the Secret makes you loose the advantage of having to remember a single field to work with while doing things declaratively (which is what my suggestion was aiming to improve).

Reply:
> makes you loose the advantage of having to remember a single field to work with while doing things declaratively / without a CLI or UI

That's a fair point.

This is a unique area of the codebase because we're trying superimpose structure and a nicer UX on top of something that is inherently very generic (Secrets). It seems we can either optimize for the portion of the experience we fully control (Kargo CLI and UI) or for the part of it we don't -- i.e. `kubectl` or GitOps'ing a project, where you are forced to work with naked Secrets.

We need to make a decision and I am very narrowly ruling in favor of optimizing for the part we control. By a narrow margin, I think it's the option that keeps more logic out of the clients and just in the back end.

Also, I think people working with naked Secrets are going to end up having to consult docs regardless, so not knowing off the top of their heads that there's a separate field to indicated "regexiness" isn't likely to be a huge problem.

Reply:
> Also, I think people working with naked Secrets are going to end up having to consult docs regardless, so not knowing off the top of their heads that there's a separate field to indicated "regexiness" isn't likely to be a huge problem.

I think in combination with the subscription filtering feature, it's quite easy to remember you have to just prefix something with `regex:` once you have used it once or twice. Which is why I am generally in favor of "pick one way of doing things and stick with it".

But I come from a world where Kubernetes API design (and the YAML) was the core foundation of everything we built, so my opinion/view may differ wildly here from Kargo's world where things tend to be more CLI/UI driven.